transport/gdrive-transport/src/main/java/org/apache/airavata/mft/transport/gdrive/GDriveSender.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Arrays;
+import java.util.Collection;
+
+
+public class GDriveSender implements Connector {
+    private static final Logger logger = LoggerFactory.getLogger(GDriveSender.class);
+
+    private GDriveResource gdriveResource;
+    private Drive drive;
+    private JsonObject jsonObject;
+
+
+    @Override
+    public void init(String resourceId, String credentialToken, String resourceServiceHost, int resourceServicePort, String secretServiceHost, int secretServicePort) throws Exception {
+        ResourceServiceGrpc.ResourceServiceBlockingStub resourceClient = ResourceServiceClient.buildClient(resourceServiceHost, resourceServicePort);
+        this.gdriveResource = resourceClient.getGDriveResource(GDriveResourceGetRequest.newBuilder().setResourceId(resourceId).build());
+
+        SecretServiceGrpc.SecretServiceBlockingStub secretClient = SecretServiceClient.buildClient(secretServiceHost, secretServicePort);
+        GDriveSecret gdriveSecret = secretClient.getGDriveSecret(GDriveSecretGetRequest.newBuilder().setSecretId(credentialToken).build());
+
+        HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = new JacksonFactory();
+        String jsonString = gdriveSecret.getCredentialsJson();
+        jsonObject = new JsonParser().parse(jsonString).getAsJsonObject();
+        GoogleCredential credential = GoogleCredential.fromStream(new ByteArrayInputStream(jsonString.getBytes(StandardCharsets.UTF_8)), transport, jsonFactory);
+
+        if (credential.createScopedRequired()) {
+            Collection<String> scopes = DriveScopes.all();
+            //Arrays.asList(DriveScopes.DRIVE,"https://www.googleapis.com/auth/drive");
+            credential = credential.createScoped(scopes);
+
+        }
+
+        drive = new Drive.Builder(transport, jsonFactory, credential)
+                .setApplicationName("My Project").build();
+
+    }
+
+    @Override
+    public void destroy() {
+
+    }
+
+    @Override
+    public void startStream(ConnectorContext context) throws Exception {
+        logger.info("Starting GDrive Sender stream for transfer {}", context.getTransferId());
+        logger.info("Content length for transfer {} {}", context.getTransferId(), context.getMetadata().getResourceSize());
+        String id = null;
+
+        InputStreamContent contentStream = new InputStreamContent(
+                "", context.getStreamBuffer().getInputStream());
+
+        String entityUser = jsonObject.get("client_email").getAsString();
+        File fileMetadata = new File();
+        fileMetadata.setName(this.gdriveResource.getResourcePath());
+
+        boolean fileupdated = false;
+        FileList fileList = drive.files().list().setFields("files(id,name)").execute();
+        logger.info("gdriveResource.getResourcePath() " + gdriveResource.getResourcePath());
+        logger.info("Listing files in GDRIVE SENDER " + drive.files().list().setFields("files(id,name)").execute());
```

Comment:
Make log lines consistent. Don't use upper cased words in middle if you don't have a very good reason. And drive.files().list().setFields("files(id,name)").execute() is a very expensive call to do in log lines. Can't you use fileList as above? Further I don't see a value in printing all file list in an info log line because it's not relevant to common monitoring. You can make this line as a debug log.

Reply:
These have been removed since they were only necessary for the development phase.